var/spack/repos/builtin/packages/intel-oneapi-mpi/package.py

```diff
@@ -200,11 +200,17 @@ def setup_dependent_package(self, module, dep_spec):
 
     def setup_dependent_build_environment(self, env, dependent_spec):
         dependent_module = dependent_spec.package.module
-        env.set("I_MPI_CC", dependent_module.spack_cc)
-        env.set("I_MPI_CXX", dependent_module.spack_cxx)
-        env.set("I_MPI_F77", dependent_module.spack_f77)
-        env.set("I_MPI_F90", dependent_module.spack_fc)
-        env.set("I_MPI_FC", dependent_module.spack_fc)
+        for var_name, attr_name in (
+            ("I_MPI_CC", "spack_cc"),
+            ("I_MPI_CXX", "spack_cxx"),
+            ("I_MPI_FC", "spack_fc"),
+            ("I_MPI_F90", "spack_fc"),
+            ("I_MPI_F77", "spack_f77"),
+        ):
+            if not hasattr(dependent_module, attr_name):
+                continue
+
+            env.set(var_name, getattr(dependent_module, attr_name))
```

Comment:
```suggestion
            if hasattr(dependent_module, attr_name):
                env.set(var_name, getattr(dependent_module, attr_name))
```

Reply:
No prob doing this, but to explain the rationale. Happy path to the left, and I guess I assumed the default is to have the attribute.

Reply:
3b4ea133d2a076afb424a3f4676c9ff54e977e8d